src/components/Navigation/NavigationHeader.tsx

```diff
@@ -1,30 +1,49 @@
+"use client"
 import Link from "next/link"
-import { FaSearch } from "react-icons/fa"
+import Image from "next/image"
 import NavigationBar from "./NavigationBar"
-
+import useAuth from "../../hooks/useAuth"
+import styles from "./navigation.module.css"
+import Logo from "../../../public/images/logos/logo_tc.png"
 const NavigationHeader = () => {
+  const { token, user } = useAuth()
+
   return (
-    <div className="bg-black py-1">
-      <div className="container mx-auto flex justify-between items-center py-2">
+    <div className={`${styles.mobile_nav} ${styles.nav}`}>
+      <div className="container mx-auto flex justify-between items-center py-2 ">
         <NavigationBar />
-        <form className="flex items-center space-x-2 ml-auto">
-          <div className="relative">
-            <input
-              type="text"
-              className="py-1 px-3 pr-10 text-white border-none bg-opacity-10 bg-gray-300 rounded-lg focus:outline-none"
-              placeholder="Search..."
-            />
-            <div className="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none">
-              <FaSearch className="text-white" />
-            </div>
+        <Image
+          src={Logo}
+          alt="true crime logo"
+          width={150}
+          height={120}
+          className={styles.logo}
+        />
+        <div className="hidden md:flex md:relative items-center ">
+          <div className="ml-auto mr-4">
+            {token ? (
+              <Link href="/profile" className="text-xl text-white">
+                {user ? (
+                  <Image
```

Comment:
I suppose both `token` and `user` checks can be replaced by a single `user` or `token` check.
Maybe even a variable.
You can replace `checkMe ? element : null` with `checkMe && element`. There are small differences and your expression is "more correct" but it's also harder to read so, your call.
Example with variable
```tsx
const isLoggedIn: boolean = user && token

return(
 ...
<div className="ml-auto mr-4">
 {isLoggedIn && (
   <Link href="/profile" className="text-xl text-white">
     <Image
        src={user.avatar}
        alt="User Avatar"
        width={50}
        height={50}
        className="rounded-full border-2" 
    />
  </Link>
...
)
```

Seems redundant? 
You can make it so that `useAuth` gives you the user or `null` if not logged in. Then you don't need both token and user object but just user object to check. In that case replace `isLoggedIn` with `user`